recipe/gen_patch_json.py

```diff
@@ -1290,6 +1290,13 @@ def _gen_new_index_per_key(repodata, subdir, index_key):
             elif record["version"] == "0.7.14":
                 _replace_pin("python >=2.7", "python >=2.7,<3.10", deps, record)
 
+        # Retroactively pin hloviews to Python < 3.10 b/c they import Callable directly and that was removed in py310.
+        # This is already fixed in next verision of holoviews >1.14.7 (unrelease).
+        if record_name == "holoviews":
+            deps = record["depends"]
+            if  record['version'] <= "1.14.7":
```

Comment:
This is a straight string comparison. I think unless I am missing some magic on encodings that we'll want to cast to a version object or something before doing this.

Reply:
Yeah, I was going to do a == first and for that it would have been ok. After some investigation I decided to do all previous version. I'll do a version object. Any suggestions? I know there a few options available but I never used any of them.

Reply:
I'm pretty sure if you search this source file, there is one already imported.

Reply:
@isuruf, any suggestions? It would be nice to make it consistent. I see many different ways to do it.

Reply:
It doesn't look like @isuruf had a chance to respond.  It looks like this approach would work:
https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/blob/d94374e7e54d79a06b881fcfbce4235b5c3c41d5/recipe/gen_patch_json.py#L551-L553
So in your case:
``` python
            pversion = pkg_resources.parse_version(record['version'])
            pv_1147 = pkg_resources.parse_version('1.14.7')
            if pversion <= pv_1147:
```

Reply:
I forgot about this one. While it would be nive to standardize this in our script. This particular package doesn't really need a repodata patch. It is controversial but one can argue that 99% of the package works with Python 3.10. Just one small functionality that does not. Also, the latest version is 100% compatible, so the patch would break workflow of folks who are pinning this at the moment.